docs/fsharp-cheatsheet.md

```diff
@@ -1,523 +1,630 @@
+let decimal            = 50.0m          // signed 128-bit decimal
 
+let byte               = 'a'B           // ascii character; 97uy
 
-    let byte               = 'a'B           // ascii character; 97uy
-
-    let byteArray          = "text"B        // ascii string; [|116uy; 101uy; 120uy; 116uy|]
+let byteArray          = "text"B        // ascii string; [|116uy; 101uy; 120uy; 116uy|]
+```
 
 *Primes* (or a tick `'` at the end of a label name) are idiomatic to functional languages and are included in F#. They are part of the identifier's name and simply indicate to the developer a variation of an existing value or function. For example:
 
-    let x = 5
-    let x' = x + 1
-    let x'' = x' + 1
+```fsharp
+let x = 5
+let x' = x + 1
+let x'' = x' + 1
+```
 
 See [Literals (MS Learn)](https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/literals) for complete reference.
 
 <div id="functions"></div>
 
-## Functions
+# Functions
 
-Use the `let` keyword to define named functions.
+## `let` bindings
 
-    let add n1 n2 = n1 + n2
-    let subtract n1 n2 = n1 - n2
-    let negate num = -1 * num
-    let print num = printfn $"The number is: {num}"
+Use the `let` keyword to define named functions.
 
-### Pipe and Composition Operators
+```fsharp
+let add n1 n2 = n1 + n2
+let subtract n1 n2 = n1 - n2
+let negate num = -1 * num
+let print num = printfn $"The number is: {num}"
+```
+## Pipe and Composition Operators
 
 Pipe operator `|>` is used to chain functions and arguments together.
 
-    let addTwoSubtractTwoNegateAndPrint num =
-        num |> add 2 |> subtract 2 |> negate |> print
+```fsharp
+let addTwoSubtractTwoNegateAndPrint num =
+    num |> add 2 |> subtract 2 |> negate |> print
+```
 
 Composition operator `>>` is used to compose functions:
 
-    let addTwoSubtractTwoNegateAndPrint' =
-        add 2 >> subtract 2 >> negate >> print
+```fsharp
+let addTwoSubtractTwoNegateAndPrint' =
+    add 2 >> subtract 2 >> negate >> print
+```
 
 Caution: The output is the _last_ argument to the next function.
 
-    // `addTwoSubtractTwoNegateAndPrint 10` becomes:
-    10
-    |> add 2       //  2 + 10  = 12
-    |> subtract 2  //  2 - 12  = -10
-    |> negate      // -1 * -10 = 10
-    |> print       // "The number is 10"
+```fsharp
+// `addTwoSubtractTwoNegateAndPrint 10` becomes:
+10
+|> add 2       //  2 + 10  = 12
+|> subtract 2  //  2 - 12  = -10
+|> negate      // -1 * -10 = 10
+|> print       // "The number is 10"
+```
+
+## Anonymous functions
+
+Anonymous, or "lambda" functions, are denoted by the `fun` keyword and the arrow operator `->`.
+
+```fsharp
+let isDescending xs =
+    xs
+    |> List.pairwise
+    |> List.forAll (fun (x, y) -> x > y)
+
+let suspiciousRecords =
+    records
+    |> Seq.filter (fun x -> x.Age >= 150)
+```
+### _.Property shorthand
+
+If the lambda function has a single argument which is used in an atomic expression, the following shorthand has been available since F# 8:
```

Comment:
```suggestion
If the lambda function has a single argument that is used in an atomic expression, the following shorthand has been available since F# 8:
```